src/site/content/en/blog/javascript-this/index.md

```diff
@@ -0,0 +1,362 @@
+
+## If the function is defined as an arrow function: {: #arrow-functions }
+
+```js
+const arrowFunction = () => {
+  console.log(this);
+};
+```
+
+In this case, the value of `this` is _always_ the same as `this` in the parent scope:
+
+```js
+const outerThis = this;
+
+const arrowFunction = () => {
+  // Always logs `true`:
+  console.log(this === outerThis);
+};
+```
+
+Arrow functions are great because the inner value of `this` can't be changed, it's _always_ the same as the outer `this`.
+
+### Other examples
+
+With arrow functions, the value of `this` _can't_ be changed with [`bind`](#bound):
+
+```js
+// Logs `true` - bound `this` value is ignored:
+arrowFunction.bind({foo: 'bar'})();
+```
+
+With arrow functions, the value of `this` _can't_ be changed with [`call` or `apply`](#call-apply):
+
+```js
+// Logs `true` - called `this` value is ignored:
+arrowFunction.call({foo: 'bar'});
+// Logs `true` - applied `this` value is ignored:
+arrowFunction.apply({foo: 'bar'});
+```
+
+With arrow functions, the value of `this` _can't_ be changed by calling the function as a member of another object:
+
+```js
+const obj = {arrowFunction};
+// Logs `true` - parent object is ignored:
+obj.arrowFunction();
+```
+
+With arrow functions, the value of `this` _can't_ be changed by calling the function as a constructor:
+
+```js
+// TypeError: arrowFunction is not a constructor
+new arrowFunction();
+```
+
+### 'Bound' instance methods
+
+With instance methods, if you want to ensure `this` always refers to the class instance, the best way is to use arrow functions and [class fields](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Public_class_fields):
+
+```js
+class Whatever {
+  someMethod = () => {
+    // Always the instance of Whatever:
+    console.log(this);
+  };
+}
+```
+
+This pattern is really useful when using instance methods as event listeners in components (such as React components, or web components).
+
+The above might feel like it's breaking the "`this` will be the same as `this` in the parent scope" rule, but it starts to make sense if you think of class fields as syntactic sugar for setting things in the constructor:
+
+```js
+class Whatever {
+  someMethod = (() => {
+    const outerThis = this;
+    return () => {
+      // Always logs `true`:
+      console.log(this === outerThis);
+    };
+  })();
+}
+
+// …is roughly equivalent to:
+
+class Whatever {
+  constructor() {
+    const outerThis = this;
+    this.someMethod = () => {
+      // Always logs `true`:
+      console.log(this === outerThis);
+    };
+  }
+}
+```
+
+## Otherwise, if the function/class is called with `new`: {: #new }
+
+```js
+new Whatever();
+```
+
+The above will call `Whatever` (or its constructor function if it's a class) with `this` set to the result of `Object.create(Whatever.prototype)`.
+
+```js
+class MyClass {
+  constructor() {
+    console.log(
+      this.constructor === Object.create(MyClass.prototype).constructor,
```

Comment:
```suggestion
      this.constructor === Object.create(MyClass.prototype).constructor
```

It's probably prettier that did this, but it really looks weird here I think.

Reply:
Yeah, I've been sticking to the prettier rules, and I disagree with the lot of them (especially the way we diverge from the defaults), but it felt right to stick to the rules.